openpdf/src/main/java/com/lowagie/text/pdf/PdfNumber.java

```diff
@@ -170,4 +172,25 @@ public void increment() {
         value += 1.0;
         setContent(ByteBuffer.formatDouble(value));
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
Also, since `PdfNumber` is not used as superclass, you can make it `final` and replace this `if` check with single `if(!(o instanceof PdfNumber))`.
And (optionally) add code block brackets for one-line expressions - it's very hard to comprehend code without proper alignment

Reply:
I agree for the `if`. 
For the `final`, do you mean that I should mark the class as final?

Reply:
Yes, that is what I meant

Reply:
I'm not sure about this, since this is a library and we can't take as granted that any user hasn't inherited from this class.